dotnet/src/SemanticKernel.Abstractions/Filters/Function/FunctionFilterCollection.cs

```diff
@@ -0,0 +1,142 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Provides a collection of <see cref="IFunctionFilter"/> instances.
+/// </summary>
+[Experimental("SKEXP0005")]
+public sealed class FunctionFilterCollection : IList<IFunctionFilter>, IReadOnlyList<IFunctionFilter>
```

Comment:
This shouldn't be public. It should only be surfaced as an `IList<IFunctionFilter>`.